src/gc-arena/src/gc_cell.rs

```diff
@@ -23,7 +23,30 @@ impl<'gc, T: Collect + 'gc> Clone for GcCell<'gc, T> {
 
 impl<'gc, T: 'gc + Collect + Debug> Debug for GcCell<'gc, T> {
     fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
-        fmt.debug_tuple("GcCell").field(&self.0).finish()
+        match self.try_read() {
+            Ok(borrow) => fmt.debug_struct("GcCell").field("value", &borrow).finish(),
+            Err(_) => {
+                // The GcCell is mutably borrowed so we can't look at its value
+                // here. Show a placeholder instead.
+                struct BorrowedPlaceholder;
+
+                impl Debug for BorrowedPlaceholder {
+                    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                        f.write_str("<borrowed>")
+                    }
+                }
+
+                fmt.debug_struct("GcCell")
+                    .field("value", &BorrowedPlaceholder)
+                    .finish()
+            }
```

Comment:
One alternative here is to display the pointer value instead of just printing '<borrowed>', which at least provides a bit more info.  Thoughts?

Reply:
I am totally fine with "<borrowed>" personally, especially if that's what the standard library currently does for Rc

Reply:
Here's what the [standard library does for reference](https://doc.rust-lang.org/src/core/fmt/mod.rs.html#2618), so I'm inclined to just leave it as it is.